### ISSUE TITLE
rocfft_aot_helper rocfft_kernel_cache.db : Assertion `thread != NULL && "cannot lock() from (null)"' failed

### ISSUE THREAD

**Issue by perestoronin** [ROCmSoftwarePlatform/rocFFT#389]
### What is the expected behavior
-   successful build

### What actually happens
```
-  [61/389] cd /var/tmp/portage/sci-libs/rocFFT-5.4.0/work/rocFFT-rocm-5.4.0_build/library/src && /var/tmp/portage/sci-libs/rocFFT-5.4.0/work/rocFFT-rocm-5.4.0_build/library/src/rocfft_aot_helper /var/tmp/portage/sci-libs/rocFFT-5.4.0/work/rocFFT-rocm-5.4.0_build/library/src/rocfft_kernel_cache.db /var/tmp/portage/sci-libs/rocFFT-5.4.0/work/rocFFT-rocm-5.4.0_build/library/src/rocfft_rtc_helper gfx900
FAILED: library/src/rocfft_kernel_cache.db /var/tmp/portage/sci-libs/rocFFT-5.4.0/work/rocFFT-rocm-5.4.0_build/library/src/rocfft_kernel_cache.db 
cd /var/tmp/portage/sci-libs/rocFFT-5.4.0/work/rocFFT-rocm-5.4.0_build/library/src && /var/tmp/portage/sci-libs/rocFFT-5.4.0/work/rocFFT-rocm-5.4.0_build/library/src/rocfft_aot_helper /var/tmp/portage/sci-libs/rocFFT-5.4.0/work/rocFFT-rocm-5.4.0_build/library/src/rocfft_kernel_cache.db /var/tmp/portage/sci-libs/rocFFT-5.4.0/work/rocFFT-rocm-5.4.0_build/library/src/rocfft_rtc_helper gfx900
rocfft_aot_helper: /var/tmp/portage/dev-util/hip-5.4.0/work/ROCclr-rocm-5.4.0/cmake/../thread/monitor.hpp:180: bool amd::Monitor::tryLock(): Assertion `thread != NULL && "cannot lock() from (null)"' failed.
```

### How to reproduce
-  try build rocFFT 5.4.0 from source

### Environment
| Hardware | description |
|-----|-----|
| GPU | Radeon Vega Frontier Edition  |
| CPU |  AMD Ryzen 9 3900XT |

| Software | version |
|-----|-----|
| ROCK | v5.4.0 |
| ROCR | v5.4.0 |
| HCC | v5.4.0 |
| Library | v5.4.0 |

PS. rocFFT 5.3.3 build succesfull, but 5.4.0 not.

**Comment by perestoronin** [comment]
With env var export ROCFFT_RTC_PROCESS=1 https://github.com/ROCmSoftwarePlatform/rocFFT/blob/rocm-5.4.0/library/src/rtc_cache.cpp#L377 case pass away.

**Comment by lubyagin** [comment]
You can to set `AMD_LOG_LEVEL = LOG_DEBUG` next time to debug this code.

**Comment by evetsso** [comment]
Hi @perestoronin , thanks for the bug report.

This is ultimately a problem in hipRTC.  As you've discovered, the ROCFFT_RTC_PROCESS=1 workaround will help get rocFFT built.

Once built, rocFFT may run into the same problem at runtime, if it needs to compile any kernels.  The same workaround would be possible there too.

We'll aim to get the underlying hipRTC issue fixed in ROCm 5.4.1.
